packages/venia-concept/src/components/SwatchTooltip/swatchTooltip.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import { shape, string } from 'prop-types';
+import over from 'lodash.over';
+import uuid from 'uuid/v4';
+
+import classify from 'src/classify';
+import defaultClasses from './swatchTooltip.css';
+
+class SwatchTooltip extends Component {
+    static propTypes = {
+        classes: shape({
+            root: string,
+            tooltip: string
+        }),
+        text: string
+    };
+
+    constructor(...args) {
+        super(...args);
+
+        this.timeoutId = null;
+        this.uniqueId = uuid();
+    }
+
+    state = {
+        isShowing: false
+    };
+
+    /*
+     *  Close the tooltip on the next tick.
+     *  Unfortunately this is necessary because another child of ours may have received focus
+     *  and this blur event is fired before that new focus event.
+     *  @see https://reactjs.org/docs/accessibility.html#mouse-and-pointer-events.
+     */
+    onBlur = () => {
+        this.timeoutId = setTimeout(() => {
+            this.setState({ isShowing: false });
+        }, 0);
+    };
+
+    /*
+     *  The tooltip target has received focus, show the tooltip.
+     */
+    onFocus = () => {
+        // If a child receives focus, do not close the tooltip yet.
+        clearTimeout(this.timeoutId);
+
+        this.setState({ isShowing: true });
+    };
+
+    onKeyDown = event => {
+        if (event.key === 'Escape') {
+            this.setState({ isShowing: false });
+        }
+    };
+    onMouseOver = () => {
+        this.setState({ isShowing: true });
+    };
+    onMouseLeave = () => {
+        this.setState({ isShowing: false });
+    };
+
+    render() {
+        const { text, children, classes } = this.props;
+        const { isShowing } = this.state;
+
+        const ariaEnhancedChildren = React.Children.map(children, child => {
+            if (child.type !== 'button') {
+                return child;
+            }
+
+            // The button triggers the tooltip.
+            // Update its accessibility accordingly.
+            const ariaEnhancedButton = React.cloneElement(child, {
+                'aria-describedby': this.uniqueId,
+                className: `${child.props.className} ${classes.trigger}`,
```

Comment:
I'm a little hesitant about composing the classnames together like this, but I'm not sure why. Basically I just want to add this component's class onto whatever is already present.

For example, `root tile` becomes `root tile trigger`.

Reply:
I'm a little confused about what the desired outcome is here. In practice this will check for the `classes.trigger` prop from the css we have created, except that in `swatchTooltip.css` there is no `trigger` class so this is `undefined`. Do we even need this class addition?

Reply:
Nice catch, thanks!

If I remember correctly it was adding the `margin` around the swatch, which I thought definitely didn't belong in the `SwatchTooltip`, so I refactored the CSS and the `trigger` class doesn't exist anymore.

I'll remove this.